latte/models/probabilistic_pca/probabilistic_pca.py

```diff
@@ -0,0 +1,196 @@
+    quality of the fit subspace.
+
+    Args:
+        N: Number of samples to generate.
+        n: The dimension of the observable space.
+        d: The dimension of the linear subspace which the data is projected on.
+        d_measured: The number of generative factors we are interested in, or we assume have measurements of.
+        spread: The standard deviation of the Gaussian used to generate the observations.
+        sigma: The standard deviation of the noise added to the observations
+               to make the discovery of the subspace harder.
+        rng: The random number generator for the dataset.
+
+    Returns:
+        Populated GaussianGenerativeDataset
+    """
+
+    rng = np.random.default_rng(rng)
+
+    Z = rng.normal(scale=spread, size=(N, d))  # The generative factors
+
+    A = rng.normal(size=(n, d))  # The mixing/loading matrix; it can be any `d x d` matrix
+    mu = rng.normal(size=(n,))
+    epsilon = rng.normal(scale=sigma, size=(N, n))
+
+    X = Z @ A.T + mu + epsilon  # The observables
+
+    # Only return the values of the factors we are interested in, not all used in the generation
+    measured_factors = np.sort(rng.choice(d, size=d_measured, replace=False))
+
+    return ProbabilisticPCADataset(
+        X=X.astype(np.float32),
+        Z=Z.astype(np.float32),
+        A=A.astype(np.float32),
+        mu=mu.astype(np.float32),
+        sigma=sigma,
+        n=n,
+        d=d,
+        d_measured=d_measured,
+        measured_factors=measured_factors,
+    )
+
+
+def gaussian_log_likelihood(X: np.ndarray, A_hat: np.ndarray, mu_hat: np.ndarray, sigma_hat: float) -> float:
+    """
+    Calculates the log-likelihood of the parameters `mu_hat1`, `sigma_hat`, and `A_hat` as the parameters of a
+    probabilistic PCA model given the data `X`.
+    Args:
+        X: The observed data
+        A_hat: The mixing matrix
+        mu_hat: The mean of the data
+        sigma_hat: The standard deviation of the observation model
+
+    Returns:
+        The log-likelihood
+    """
+    cov_hat = A_hat @ A_hat.T + sigma_hat**2 * np.eye(A_hat.shape[0])
+    return stats.multivariate_normal(mean=mu_hat, cov=cov_hat).logpdf(X).sum()
+
+
+def generative_model_mutual_information(A: np.ndarray, sigma: float) -> float:
+    """
+    Analytically computes the mutual information between the latent and the observable vectors from the probabilistic
+    PCA generative model with the mixing matrix A.
+    Args:
+        A: The model mixing matrix
+        sigma: The variance of the observable model
+
+    Returns:
+        The true mutual information between the subvectors.
+    """
+
+    n, d = A.shape
+    sigma = np.vstack((np.hstack((np.eye(d), A.T)), np.hstack((A, A @ A.T + sigma**2 * np.eye(n)))))
+
+    return 1 / 2 * np.log(np.linalg.det(sigma[:d, :d]) * np.linalg.det(sigma[d:, d:]) / np.linalg.det(sigma))
+
+
+def gaussian_mutual_information(sigma: np.ndarray, d: int) -> float:
```

Comment:
Nice! We have a utility for this in the KSG estimator tests as well. (Probably at some stage we'll merge these, but let's not worry about it before the deadline).